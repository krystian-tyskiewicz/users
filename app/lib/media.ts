const media = {
  phone: '@media (max-width: 480px)',
  tablet: '@media (max-width: 768px)',
}

export default media
