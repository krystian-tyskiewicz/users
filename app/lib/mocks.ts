export const user = {
  id: '1234',
  name: 'John Doe',
  username: 'john-doe',
  email: 'john@doe.com',
  city: 'New York',
}
