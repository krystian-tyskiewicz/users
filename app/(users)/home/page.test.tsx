import { render } from '@testing-library/react'

import Page from './page'
import { user } from '@/app/lib/mocks'

jest.mock('../../redux/services/users', () => ({
  useGetUsersQuery: jest.fn().mockImplementation(() => ({
    isLoading: false,
    data: [user],
  })),
  useDeleteUserMutation: jest.fn().mockReturnValue([])
}))
jest.mock('next/navigation', () => ({
  useRouter: jest.fn(),
}))
jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}))

it('renders correctly', () => {
  const { container } = render(<Page />)
  expect(container).toMatchSnapshot()
})
