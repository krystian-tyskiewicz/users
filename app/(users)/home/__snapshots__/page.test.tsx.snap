// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <header
    class="sc-jRQBzY bPfapj"
  >
    <h4
      class="sc-bqiRFl eSiAwy"
    >
      Users List
    </h4>
    <button
      class="sc-bdvumH haoqpZ"
      type="button"
    >
      Add new
    </button>
  </header>
  <div
    class="sc-fFeiqI elCEmB"
  >
    <table
      class="sc-pVSyH hFSsQO"
    >
      <thead
        class="sc-jrQzhe"
      >
        <tr
          class="sc-iqsfQQ"
        >
          <th
            class="sc-crHkVz fpdoEl"
          >
            Id
          </th>
          <th
            class="sc-crHkVz fpdoEl"
          >
            Name
          </th>
          <th
            class="sc-crHkVz fpdoEl"
          >
            Username â†“
          </th>
          <th
            class="sc-crHkVz fpdoEl"
          >
            Email
          </th>
          <th
            class="sc-crHkVz fpdoEl"
          >
            City
          </th>
          <th
            class="sc-crHkVz fpdoEl"
          >
            Edit
          </th>
          <th
            class="sc-crHkVz fpdoEl"
          >
            Delete
          </th>
        </tr>
      </thead>
      <tbody
        class="sc-kDTiHp"
      >
        <tr
          class="sc-iqsfQQ"
        >
          <td
            class="sc-egifa chFxGh"
          >
            1234
          </td>
          <td
            class="sc-egifa chFxGh"
          >
            John Doe
          </td>
          <td
            class="sc-egifa chFxGh"
          >
            john-doe
          </td>
          <td
            class="sc-egifa chFxGh"
          >
            john@doe.com
          </td>
          <td
            class="sc-egifa chFxGh"
          >
            New York
          </td>
          <td
            class="sc-egifa chFxGh"
          >
            <button
              class="sc-bdvumH hcIvqc"
              type="button"
            >
              Edit
            </button>
          </td>
          <td
            class="sc-egifa chFxGh"
          >
            <button
              class="sc-bdvumH ynMys"
              type="button"
            >
              Delete
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
